# Patch from https://aur.archlinux.org/packages/zfs-dkms/
# Avoids recompiling ZFS userland utilities with DKMS rebuilds
# Also short-circuits a useless test for Debian in the DKMS source

--- a/scripts/dkms.mkconf	2021-01-19 19:19:08.000000000 -0500
+++ b/scripts/dkms.mkconf	2021-02-02 19:03:02.045984881 -0500
@@ -26,22 +26,7 @@
 PRE_BUILD="configure
   --prefix=/usr
   --with-config=kernel
-  --with-linux=\$(
-    case \`lsb_release -is\` in
-      (Debian|Devuan)
-        if [[ -e \${kernel_source_dir/%build/source} ]]
-        then
-          echo \${kernel_source_dir/%build/source}
-        else
-          # A kpkg exception for Proxmox 2.0
-          echo \${kernel_source_dir}
-        fi
-      ;;
-      (*)
-        echo \${kernel_source_dir}
-      ;;
-    esac
-  )
+  --with-linux=\${kernel_source_dir}
   --with-linux-obj=\${kernel_source_dir}
   \$(
     [[ -n \"\${ICP_ROOT}\" ]] && \\
@@ -75,7 +60,7 @@
 "
 AUTOINSTALL="yes"
 REMAKE_INITRD="no"
-MAKE[0]="make"
+MAKE[0]="make -C module/"
 STRIP[0]="\$(
   [[ -r \${PACKAGE_CONFIG} ]] \\
   && source \${PACKAGE_CONFIG} \\
